AudioKit/Common/Internals/Audio File/AKAudioFile+Async Processing.swift

```diff
@@ -0,0 +1,365 @@
+//
+//  AKAudioFile+processing Async.swift
+//  AudioKit For iOS
+//
+//  Created by Laurent Veliscek and Brandon Barber on 12/07/2016.
+//  Copyright © 2016 AudioKit. All rights reserved.
+//
+//
+//  IMPORTANT: Any AKAudioFile process will output a .caf AKAudioFile
+//  set with a PCM Linear Encoding (no compression)
+//  But it can be applied to any readable file (.wav, .m4a, .mp3...)
+//
+
+
+import Foundation
+import AVFoundation
+
+
+
+extension AKAudioFile {
+
+
+    // MARK: - embedded enum
+
+    /// Enum of Status returned from the status property
+    /// of any AKAudioFile Async AKAudioFile process.
+    public enum ProcessStatus {
+        case Failed
+        case Processing
+        case Succeeded
+    }
+    /// Enum of Process Types stored as a property
+    /// in any AKAudioFile Async AKAudioFile process.
+    public enum ProcessType {
+        case Normalize
+        case Reverse
+        case Append
+    }
+
+    // MARK: - Async Process Objects
+
+    /*
+    Returned from normalize_Async()
+    
+    As soon as completionHandler has been triggered,
+    you can check status (.Failed or .Succeeded)
+     
+    If .Succeeded, the resulting AKAudioFile is the
+    process.processedFile property.
+
+    If .Failed, you mau check the error thrown using
+    process.error property.
+     
+    If needed, you can get the source AKAudioFile using
+    process.sourceFile property and get the process type
+    using process.processType
+
+    */
+    public class NormalizeProcess {
```

Comment:
Public classes in AudioKit all start with AK, or should.  This is such a generic non-descriptive name too.